File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    inorderTraversal(root) {
        let array=[];
        return this.inOrder(root, array);
    }

    inOrder(node, array){
        if(node==null)
            return array;
        
        this.inOrder(node.left, array)
        array.push(node.val);
        this.inOrder(node.right,array)
        return array;
    }
}

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> res;
        stack<pair<TreeNode*,bool>> pila;
        pila.push(make_pair(root, false));

        while(!pila.empty()){
            pair<TreeNode*, bool> curr = pila.top(); pila.pop();

            if(curr.first){
                if(!curr.second){
                    pila.push(make_pair(curr.first, true));
                    pila.push(make_pair(curr.first->right, false));
                    pila.push(make_pair(curr.first->left, false));
                } else {
                    res.push_back(curr.first->val);
                }
            }
            
        }

        return res;
        
    }
};